packages/odyssey-react-mui/src/labs/TopNav.tsx

```diff
@@ -0,0 +1,430 @@
+    () => ({
+      paddingRight: odysseyDesignTokens.Spacing2,
+    }),
+    [odysseyDesignTokens],
+  );
+
+  const userProfileInfoContainerStyles = useMemo(
+    () => ({
+      display: "flex",
+      flexDirection: "column",
+    }),
+    [],
+  );
+
+  const userProfileEmailInfoStyles = useMemo(
+    () => ({
+      color: odysseyDesignTokens.TypographyColorHeading,
+      fontSize: odysseyDesignTokens.TypographySizeSubordinate,
+    }),
+    [odysseyDesignTokens],
+  );
+
+  const userProfileOrgInfoStyles = useMemo(
+    () => ({
+      color: odysseyDesignTokens.TypographyColorSubordinate,
+      fontSize: odysseyDesignTokens.TypographySizeSubordinate,
+    }),
+    [odysseyDesignTokens],
+  );
+
+  return (
+    <UserProfileContainer odysseyDesignTokens={odysseyDesignTokens}>
+      {profileIcon && <Box sx={userProfileIconStyles}>{profileIcon}</Box>}
+      <Box sx={userProfileInfoContainerStyles}>
+        <Box sx={userProfileEmailInfoStyles}>{userName}</Box>
+        <Box sx={userProfileOrgInfoStyles}>{orgName}</Box>
+      </Box>
+    </UserProfileContainer>
+  );
+};
+
+const TopNavListContainer = styled.ul({
+  padding: 0,
+  listStyle: "none",
+  listStyleType: "none",
+  display: "flex",
+  alignItems: "center",
+});
+
+const TopNavItemLabelContainer = styled("div", {
+  shouldForwardProp: (prop) => prop !== "odysseyDesignTokens",
+})<{
+  odysseyDesignTokens: DesignTokens;
+}>(({ odysseyDesignTokens }) => ({
+  display: "flex",
+  flexWrap: "wrap",
+  alignItems: "center",
+  fontSize: odysseyDesignTokens.TypographyScale0,
+  fontWeight: odysseyDesignTokens.TypographyWeightHeading,
+}));
+
+const TopNavListItemContainer = styled("li", {
+  shouldForwardProp: (prop) =>
+    prop !== "odysseyDesignTokens" && prop !== "isDisabled",
+})<{
+  odysseyDesignTokens: DesignTokens;
+  disabled?: boolean;
+  isDisabled?: boolean;
+}>(({ odysseyDesignTokens, isDisabled }) => ({
+  display: "flex",
+  alignItems: "center",
+  cursor: isDisabled ? "default" : "pointer",
+  pointerEvents: isDisabled ? "none" : "auto",
+  "& a": {
+    display: "flex",
+    alignItems: "center",
+    padding: `${odysseyDesignTokens.Spacing2} ${odysseyDesignTokens.Spacing4}`,
+    color: `${odysseyDesignTokens.TypographyColorHeading} !important`,
+  },
+  "& a:hover": {
+    textDecoration: "none",
+    backgroundColor: !isDisabled ? odysseyDesignTokens.HueNeutral50 : "inherit",
+  },
+  "& div[role='button']:hover": {
+    backgroundColor: !isDisabled ? odysseyDesignTokens.HueNeutral50 : "inherit",
+  },
+  "& a:focus-visible": {
+    outlineOffset: 0,
+    borderRadius: 0,
+    outlineWidth: odysseyDesignTokens.FocusOutlineWidthMain,
+    backgroundColor: !isDisabled ? odysseyDesignTokens.HueNeutral50 : "inherit",
+  },
+}));
+
+const TopNavItemContent = ({
+  id,
+  label,
+  href,
+  target,
+  onClick,
+  isDisabled,
+}: TopNavLinkItem) => {
+  const odysseyDesignTokens = useOdysseyDesignTokens();
+
+  const NavItemContentClickContainer = styled("div", {
```

Comment:
We can probably pull this out of the component right? Just pass it the `isDisabled` prop?